### PR TITLE
Update to use AWSNativeSDK 1.9.50

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
* Update 3rdparty windows cmake to use AWSNativeSDK 1.9.50
* Update AWSCore unit test as SDK update requires SDK init to be called in test setup step

## Testing
Tested locally: monolithic and non-monolithic build with debug, profile and release different config

### CODE REVIEW

File: Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSNativeSDKInit.h [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #include <AzCore/Module/Environment.h>
 
 #if defined(PLATFORM_SUPPORTS_AWS_NATIVE_SDK)
-
+#include <AzCore/PlatformIncl.h>
```
Comment (junbo75) [inline Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSNativeSDKInit.h:16]: Nit: Should this header file be included inside or outside #if defined(PLATFORM_SUPPORTS_AWS_NATIVE_SDK)?

In Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSMemoryInterface.h, seems like it's included outside #if defined(PLATFORM_SUPPORTS_AWS_NATIVE_SDK).

I'm curious about the reason to include this header as well.

File: Gems/AWSCore/Code/Tests/Configuration/AWSCoreConfigurationTest.cpp [hunk 2]
```diff
@@ -53,75 +51,26 @@ class AWSCoreConfigurationTest
...
-        CreateTestSetRegFile(TEST_VALID_RESOURCE_MAPPING_SETREG);
+        CreateFile(m_setRegFilePath.Native(), TEST_VALID_RESOURCE_MAPPING_SETREG);
+        m_localFileIO->SetAlias("@projectroot@", GetTestTempDirectoryPath().Native().c_str());
     }
 
     void TearDown() override
     {
-        RemoveTestFile();
-        RemoveTestDirectory();
+        RemoveFile(m_setRegFilePath.Native());
```
Comment (junbo75) [inline Gems/AWSCore/Code/Tests/Configuration/AWSCoreConfigurationTest.cpp:60]: Curious: Is the file created at the root? If not, is there any directory you need to clean up during TearDown() as well?

File: Gems/AWSCore/Code/Tests/ScriptCanvas/AWSScriptBehaviorS3Test.cpp [hunk 3]
```diff
@@ -145,17 +145,18 @@ TEST_F(AWSScriptBehaviorS3Test, GetObjectRaw_CallWithOutfileDirectoryNoExist_Inv
     AWSScriptBehaviorS3::GetObjectRaw("dummyBucket", "dummyObject", "dummyRegion", dummyDirectory);
 }
 
+#if defined(AZ_DEBUG_BUILD) || defined(AZ_PROFILE_BUILD)
+// The preparation step for this test case does not work in release mode
```
Comment (junbo75) [inline Gems/AWSCore/Code/Tests/ScriptCanvas/AWSScriptBehaviorS3Test.cpp:149]: Curious: Could you provide more details on this (i.e. the preparation step you mean here)? Is this a test specific issue or a bug for release build?

File: cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake [hunk 4]
```diff
@@ -26,7 +26,7 @@ ly_associate_package(PACKAGE_NAME DirectXShaderCompilerDxc-1.6.2104-o3de-rev3-wi
 ly_associate_package(PACKAGE_NAME SPIRVCross-2021.04.29-rev1-windows                    TARGETS SPIRVCross                  PACKAGE_HASH 7d601ea9d625b1d509d38bd132a1f433d7e895b16adab76bac6103567a7a6817)
 ly_associate_package(PACKAGE_NAME tiff-4.2.0.15-rev3-windows                            TARGETS TIFF                        PACKAGE_HASH c6000a906e6d2a0816b652e93dfbeab41c9ed73cdd5a613acd53e553d0510b60)
 ly_associate_package(PACKAGE_NAME freetype-2.10.4.16-windows                            TARGETS freetype                    PACKAGE_HASH 9809255f1c59b07875097aa8d8c6c21c97c47a31fb35e30f2bb93188e99a85ff)
-ly_associate_package(PACKAGE_NAME AWSNativeSDK-1.7.167-rev4-windows                     TARGETS AWSNativeSDK                PACKAGE_HASH a900e80f7259e43aed5c847afee2599ada37f29db70505481397675bcbb6c76c)
+ly_associate_package(PACKAGE_NAME AWSNativeSDK-1.9.50-rev2-windows                      TARGETS AWSNativeSDK                PACKAGE_HASH 047de23fa57d33196666c22f45afc9c628bae354a6c39d774cbeee8054b2eb53)
```
Comment (junbo75) [inline cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake:29]: Curious: Why is this package rev2 instead of rev1?

Comment (junbo75) [reply cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake:29]: Never mind. Saw your changes in another PR: https://github.com/o3de/3p-package-source/pull/74/commits/176bee726e5e753ddeaa870f23f62479aafc1bcb

File: Gems/AWSCore/Code/Tests/Configuration/AWSCoreConfigurationTest.cpp (hunk 2 above)
Comment (onecent1101) [reply Gems/AWSCore/Code/Tests/Configuration/AWSCoreConfigurationTest.cpp:60]: file created in test temp folder which is located in system temp folder and automatically gets cleaned up. Refer to AZ::Test::ScopedAutoTempDirectory

Comment (junbo75) [reply Gems/AWSCore/Code/Tests/Configuration/AWSCoreConfigurationTest.cpp:60]: Sounds good.

File: Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSNativeSDKInit.h (hunk 1 above)
Comment (onecent1101) [reply Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSNativeSDKInit.h:16]: strictly speaking the header should be included inside #if, but we don't really have practical use case that AWSNativeSDK is in used but PLATFORM_SUPPORTS_AWS_NATIVE_SDK is not defined, if it happens then these files will fail the build as implementation requires to include aws header.

And the reason to include this header is to resolve conflict function name with windows header, o3de has undefined several conflict function or macros for different platforms. It is a common issue, and our problem refers to GetObject https://github.com/Tencent/rapidjson/issues/1448

File: Gems/AWSCore/Code/Tests/ScriptCanvas/AWSScriptBehaviorS3Test.cpp (hunk 3 above)
Comment (onecent1101) [reply Gems/AWSCore/Code/Tests/ScriptCanvas/AWSScriptBehaviorS3Test.cpp:149]: It is a test specific issue, try to create a read-only file for this test case in setup step, as we want to test write into a read-only file. But seems like in release version, the read-only file is failed to get created so just disable this test case for release build. (as unit test cover the functionality check in debug and profile build)

**Review by lawsonamzn — APPROVED** [review]

**Review by junbo75 — APPROVED** [review]

**Pull request merged**